File: .claude/hooks/session-end-cleanup.mjs
#!/usr/bin/env node
// @hookstack session-end-cleanup-temp
// Nettoie les fichiers temporaires Claude datant de plus de 24h (SessionEnd)
import { readdirSync, statSync, unlinkSync } from "node:fs";
import { join } from "node:path";
import { fileURLToPath } from "node:url";

const MAX_AGE_MS = 24 * 60 * 60 * 1_000;
const PREFIX = "claude-";

export function run({
	readdir = readdirSync,
	stat = statSync,
	unlink = unlinkSync,
	tmp = "/tmp",
	maxAge = MAX_AGE_MS,
	now = () => Date.now(),
} = {}) {
	let cleaned = 0;
	try {
		for (const f of readdir(tmp)) {
			if (!f.startsWith(PREFIX)) continue;
			const fp = join(tmp, f);
			try {
				const age = now() - stat(fp).mtimeMs;
				if (age > maxAge) {
					unlink(fp);
					cleaned++;
				}
			} catch {}
		}
	} catch {}

	return cleaned > 0
		? {
				cleaned,
				message: `[session-end-cleanup] ${cleaned} fichier(s) temporaire(s) supprimé(s).\n`,
			}
		: { cleaned: 0 };
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	const result = run();
	if (result?.message) process.stderr.write(result.message);
}

File: .claude/hooks/session-end-cleanup.py
#!/usr/bin/env python3
# @hookstack session-end-cleanup-temp
"""Cleans up Claude temp files older than 24h (SessionEnd)."""
import os
import sys
import time

MAX_AGE_SECONDS = 24 * 60 * 60
PREFIX = "claude-"


def run(
    *,
    listdir=None,
    stat=None,
    unlink=None,
    tmp="/tmp",
    max_age=MAX_AGE_SECONDS,
    now=None,
):
    if listdir is None:
        listdir = os.listdir
    if stat is None:
        stat = os.stat
    if unlink is None:
        unlink = os.unlink
    if now is None:
        now = time.time

    cleaned = 0
    try:
        for name in listdir(tmp):
            if not name.startswith(PREFIX):
                continue
            path = os.path.join(tmp, name)
            try:
                age = now() - stat(path).st_mtime
                if age > max_age:
                    unlink(path)
                    cleaned += 1
            except OSError:
                pass
    except OSError:
        pass

    if cleaned > 0:
        return {
            "cleaned": cleaned,
            "message": f"[session-end-cleanup] {cleaned} fichier(s) temporaire(s) supprimé(s).\n",
        }
    return {"cleaned": 0}


if __name__ == "__main__":
    result = run()
    if result.get("message"):
        sys.stderr.write(result["message"])
